File: Chronos.Java.BasicProfiler/source/Chronos.Java.BasicProfiler.Agent/ThreadInfo.cpp
#include "stdafx.h"
#include "Chronos.Java.BasicProfiler.Agent.h"

namespace Chronos
{
	namespace Agent
	{
		namespace Java
		{
			namespace BasicProfiler
			{
				ThreadInfo::ThreadInfo()
					: _osThreadId(0)
				{
				}
				
				void ThreadInfo::PrepareClose()
				{
					_osThreadId = GetMetadata()->GetOsThreadId();
				}

				__uint ThreadInfo::GetOsThreadId()
				{
					if (GetIsAlive() && _osThreadId == 0)
					{
						_osThreadId = GetMetadata()->GetOsThreadId();
					}
					return _osThreadId;
				}

				Reflection::ThreadMetadata* ThreadInfo::GetMetadataInternal()
				{
					Reflection::ThreadMetadata* metadata;
					jthread thread = __UID_TO_JTHREAD(Id);
					_metadataProvider->GetThread(thread, &metadata);
					return metadata;
				}

				void ThreadInfo::OnLoaded()
				{
					PrepareClose();
				}
			}
		}
	}
}

File: source/Chronos.Java.BasicProfiler/Chronos.Java.BasicProfiler/ThreadInfo.cs
using System;
using Chronos.Model;

namespace Chronos.Java.BasicProfiler
{
    [Serializable]
    public sealed class ThreadInfo : UnitBase
    {
        public ThreadInfo(ThreadNativeInfo threadInfo)
            : base(threadInfo)
        {
        }

        private ThreadNativeInfo ThreadNativeInfo
        {
            get { return (ThreadNativeInfo)NativeUnit; }
        }

        public uint OsThreadId
        {
            get { return ThreadNativeInfo.OsThreadId; }
        }

        internal void SetDependencies()
        {
        }
    }
}
